### ISSUE TITLE
Exception when passing pandas.Series after upgrade from 3.9.0 to 4.0.6

### ISSUE THREAD

**Issue by dalito** [blink1073/oct2py#125]
Calling the function [backcor.m](http://de.mathworks.com/matlabcentral/fileexchange/27429-background-correction?focused=3778803&tab=function) from a Python script on Windows 7 gives an exception with oct2py version 4.0.6 but runs fine with 3.9.0. 

Versions: Windows 7 (64bit), Python 2.7.13 (32bit), Octave 4.2.1, octave-kernel 0.26.2, scipy 0.19.0, numpy 1.12.1

Python code calling the function:
`baseline,a,it,ord,s,fct = octave.backcor(n, y, ord_, s, f)`

After reading what changed in oct2py from 3.9.0 to 4.x I thought that I should add nout=6 as argument, but the error remained.
`baseline,a,it,ord,s,fct = octave.backcor(n, y, ord_, s, f, nout=6)`

Traceback:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-17-a56b9cd11188> in <module>()
     17 for f in ['atq', 'ah']:
     18     # baseline,a,it,ord,s,fct = octave.backcor(n, y, ord_, s, f)
---> 19     baseline,a,it,ord,s,fct = octave.backcor(n, y, ord_, s, f, nout=6)
     20 

c:\dev_x86\python27\lib\site-packages\oct2py\dynamic.pyc in __call__(self, *inputs, **kwargs)
     94                         for item in pair)
     95 
---> 96         return self._ref().feval(self.name, *inputs, **kwargs)
     97 
     98     def __repr__(self):

c:\dev_x86\python27\lib\site-packages\oct2py\core.pyc in feval(self, func_path, *func_args, **kwargs)
    367         return self._feval(func_name, func_args, dname=dname, nout=nout,
    368                           timeout=timeout, stream_handler=stream_handler,
--> 369                           store_as=store_as, plot_dir=plot_dir)
    370 
    371     def eval(self, cmds, verbose=True, timeout=None, stream_handler=None,

c:\dev_x86\python27\lib\site-packages\oct2py\core.pyc in _feval(self, func_name, func_args, dname, nout, timeout, stream_handler, store_as, plot_dir)
    541 
    542         write_file(req, out_file, oned_as=self._oned_as,
--> 543                    convert_to_float=self.convert_to_float)
    544 
    545         # Set up the engine and evaluate the `_pyeval()` function.

c:\dev_x86\python27\lib\site-packages\oct2py\io.pyc in write_file(obj, path, oned_as, convert_to_float)
     44         with _WRITE_LOCK:
     45             savemat(path, data, appendmat=False, oned_as=oned_as,
---> 46                     long_field_names=True)
     47     except KeyError:  # pragma: no cover
     48         raise Exception('could not save mat file')

c:\dev_x86\python27\lib\site-packages\scipy\io\matlab\mio.pyc in savemat(file_name, mdict, appendmat, format, long_field_names, do_compression, oned_as)
    209     else:
    210         raise ValueError("Format should be '4' or '5'")
--> 211     MW.put_variables(mdict)
    212     if file_is_string:
    213         file_stream.close()

c:\dev_x86\python27\lib\site-packages\scipy\io\matlab\mio5.pyc in put_variables(self, mdict, write_header)
    844                 self.file_stream.write(out_str)
    845             else:  # not compressing
--> 846                 self._matrix_writer.write_top(var, asbytes(name), is_global)

c:\dev_x86\python27\lib\site-packages\scipy\io\matlab\mio5.pyc in write_top(self, arr, name, is_global)
    585         self._var_name = name
    586         # write the header and data
--> 587         self.write(arr)
    588 
    589     def write(self, arr):

c:\dev_x86\python27\lib\site-packages\scipy\io\matlab\mio5.pyc in write(self, arr)
    616             self.write_struct(narr)
    617         elif narr.dtype.hasobject:  # cell array
--> 618             self.write_cells(narr)
    619         elif narr.dtype.kind in ('U', 'S'):
    620             if self.unicode_strings:

c:\dev_x86\python27\lib\site-packages\scipy\io\matlab\mio5.pyc in write_cells(self, arr)
    721         A = np.atleast_2d(arr).flatten('F')
    722         for el in A:
--> 723             self.write(el)
    724 
    725     def write_empty_struct(self):

c:\dev_x86\python27\lib\site-packages\scipy\io\matlab\mio5.pyc in write(self, arr)
    614             self.write_empty_struct()
    615         elif narr.dtype.fields:  # struct array
--> 616             self.write_struct(narr)
    617         elif narr.dtype.hasobject:  # cell array
    618             self.write_cells(narr)

c:\dev_x86\python27\lib\site-packages\scipy\io\matlab\mio5.pyc in write_struct(self, arr)
    733         self.write_header(matdims(arr, self.oned_as),
    734                           mxSTRUCT_CLASS)
--> 735         self._write_items(arr)
    736 
    737     def _write_items(self, arr):

c:\dev_x86\python27\lib\site-packages\scipy\io\matlab\mio5.pyc in _write_items(self, arr)
    750         for el in A:
    751             for f in fieldnames:
--> 752                 self.write(el[f])
    753 
    754     def write_object(self, arr):

c:\dev_x86\python27\lib\site-packages\scipy\io\matlab\mio5.pyc in write(self, arr)
    606         if narr is None:
    607             raise TypeError('Could not convert %s (type %s) to array'
--> 608                             % (arr, type(arr)))
    609         if isinstance(narr, MatlabObject):
    610             self.write_object(narr)

TypeError: Could not convert None (type <type 'NoneType'>) to array
```

**Comment by blink1073** [comment]
Hi @dalito, it looks like the new parsing is getting stuck on one of the values (there is a `None` in there that isn't getting converted to another value before trying to use `matwrite` in `scipy`.  Can you provide the script used to generate the input values?

**Comment by dalito** [comment]
Making a small example was already quite helpful. I found out that the exception occurs only if I pass a pandas series to backcor. So I can easily avoid the problem by passing the array (pd.Series.values).

Here is an example that results in the above exception:

```
import numpy as np
import pandas as pd
from oct2py import octave
octave.addpath(r'path_to_backcor_directory')

ord_ = 4   # order of polynom used to model the baseline
s = 1.5     # s: threshold for cost function
n = np.linspace(0., 2., 50)  # time
y = 10. + np.random.normal(0, 1, 50) * 0.2  # signal

y = pd.Series(y)  # exception disappears if this line is commented out

for f in ['atq', 'ah']:
    baseline,a,it,ord_,s,fct = octave.backcor(n, y, ord_, s, f, nout=6)
```

**Comment by blink1073** [comment]
Glad you found a workaround.  Using pandas was not something I had tried directly.  Thanks for the example!   I'll leave this open until I have a chance to make a fix.

**Comment by dalito** [comment]
Thanks for the prompt reply. I changed the issue title to reflect the finding.

**Comment by blink1073** [comment]
Fixed in 4.2.0.
